### PR TITLE
Improvements and simplification for explosive GUIs

### PR DESCRIPTION
**Pull request by MaxIsJoe**
more work on #6149 

On a backend part, I've simplified the code a lot and removed a lot of checks to make the GUI more responsive and to avoid confusion for players when things dont work.
Also fixed status text not updating at all.

on a more visible change, I've removed the toggle switch and replaced it with a simple button as NetUI simply doesn't like that component and is very buggy when dealing with it which led a lot of people to get confused when trying to switch modes.

It's been replaced by a simple button and you'll get various different (functional) indicators like a new icon that displays if the bomb is awaiting a signal or not.

Added the ability to exit the UI by just clicking on the outside areas of it

added sounds

more feedback text to tell the player what is going on

### CODE REVIEW

File: UnityProject/Assets/Scripts/UI/Items/GUI_Explosive.cs [hunk 1]
```diff
@@ -68,62 +67,39 @@ private IEnumerator WaitForProvider()
...
 		{
-			if(modeToggleButton.Value == "1" || sbModeToggleButton.Value == "1")
-			{
-				explosiveDevice.ToggleMode(true);
-			}
-			else
+			explosiveDevice.ToggleMode(!explosiveDevice.DetonateImmediatelyOnSignal);
+			signalIcon.SetValueServer(explosiveDevice.DetonateImmediatelyOnSignal ? "0" : "1");
+			UpdateStatusText();
+			if(Peepers.Count == 0) return;
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/UI/Items/GUI_Explosive.cs]: This statement doesn't add anything.

**Review by NoooneyDude — DISMISSED** [review]

File: UnityProject/Assets/Scripts/UI/Items/GUI_Explosive.cs (hunk 1 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/UI/Items/GUI_Explosive.cs]: ...which one?

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/UI/Items/GUI_Explosive.cs]: I assume you mean `if(Peepers.Count == 0) return;` so i just removed it and fixed other stuff.

**Comment by MaxIsJoe** [comment]
approvals pls

**Review by NoooneyDude — DISMISSED** [review]

**Comment by MaxIsJoe** [comment]
Merge this already

**Comment by MaxIsJoe** [comment]
wait i think i broke it

**Comment by MaxIsJoe** [comment]
actually you know what it's fiinnnee

**Comment by MaxIsJoe** [comment]
all these are my changes so its fiiinne

**Review by corp-0 — APPROVED** [review]

**Pull request merged**